File: WEEK20/그리디_체육복.js
function solution(n, lost, reserve) {
  let _lost = lost.filter((element) => !reserve.includes(element));
  let _reserve = reserve.filter((element) => !lost.includes(element));

  _reserve.forEach((element) => {
    if (_lost.includes(element - 1)) {
      _lost.splice(_lost.indexOf(element - 1), 1);
    } else if (_lost.includes(element + 1)) {
      _lost.splice(_lost.indexOf(element + 1), 1);
    }
  });

  let answer = n - _lost.length;

  return answer;
}

function solution(n, lost, reserve) {
  let _lost = lost.filter((element) => !reserve.includes(element));
  let _reserve = reserve.filter((element) => !lost.includes(element));

  let setLost = [...new Set(_lost)];
  let setReserve = [...new Set(_reserve)];

  setReserve.forEach((element) => {
    if (setLost.includes(element - 1)) {
      setLost.splice(setLost.indexOf(element - 1), 1);
    } else if (setLost.includes(element + 1)) {
      setLost.splice(setLost.indexOf(element + 1), 1);
    }
  });
  let answer = n - setLost.length;

  return answer;
}

File: WEEK20/그리디_체육복.py
def solution(n, lost, reserve):
    _lost = set([i for i in lost if i not in reserve])
    _reserve = set([j for j in reserve if j not in lost])
    
    for num in _reserve:
        if num - 1 in _lost:
            _lost.remove(num - 1)
        elif num + 1 in _lost:
            _lost.remove(num + 1)
    
    answer = n - len(_lost)
    
    return answer